### PR TITLE
feat(medusa): Allow to query order/product by SC

### PR DESCRIPTION
**Pull request by adrien2p**
**What**

Allow to query the products and orders by sales channels

**How**

Updating the existing end points and repositories  (if necessary) to take a new query params that is sales_channel_id as an array of string

**Tests**

Add new integration tests

Fixes CORE-295
Fixes CORE-288

### CODE REVIEW

File: integration-tests/api/__tests__/admin/order.js [hunk 1]
```diff
@@ -31,8 +34,13 @@ describe("/admin/orders", () => {
 
   beforeAll(async () => {
     const cwd = path.resolve(path.join(__dirname, "..", ".."))
-    dbConnection = await initDb({ cwd })
-    medusaProcess = await setupServer({ cwd })
+    const [serverProcess, connection] = await startServerWithEnvironment({
+      cwd,
+      env: { MEDUSA_FF_SALES_CHANNELS: true },
+      verbose: false,
+    })
```
Comment (srindom) [inline integration-tests/api/__tests__/admin/order.js:41]: todo: the feature flag should only be enabled for the tests related to sales channels. So all the existing tests in this file should run under the "old" version, while only the tests that use the sales channels should have the feature flag applied.

File: integration-tests/api/__tests__/admin/product.js [hunk 2]
```diff
@@ -21,8 +24,13 @@ describe("/admin/products", () => {
...
     const cwd = path.resolve(path.join(__dirname, "..", ".."))
-    dbConnection = await initDb({ cwd })
-    medusaProcess = await setupServer({ cwd, verbose: false })
+    const [serverProcess, connection] = await startServerWithEnvironment({
+      cwd,
+      env: { MEDUSA_FF_SALES_CHANNELS: true },
+      verbose: false,
+    })
+    medusaProcess = serverProcess
+    dbConnection = connection
```
Comment (srindom) [inline integration-tests/api/__tests__/admin/product.js:33]: todo: same as above

File: integration-tests/api/__tests__/admin/product.js [hunk 3]
```diff
@@ -845,6 +853,45 @@ describe("/admin/products", () => {
...
+      }
+
+      const productSalesChannel = await simpleProductFactory(dbConnection, productSalesChannelData)
+      const salesChannel = await simpleSalesChannelFactory(dbConnection, {
+        name: "test name",
+        description: "test description",
+      })
+      await dbConnection.manager.query(`
+        INSERT INTO product_sales_channel VALUES ('${productSalesChannel.id}', '${salesChannel.id}')
+      `)
```
Comment (srindom) [inline integration-tests/api/__tests__/admin/product.js:872]: nit(non-blocking): could we maybe add a products field to the `simpleSalesChannelFactory` so it becomes this:

```
      const salesChannel = await simpleSalesChannelFactory(dbConnection, {
        name: "test name",
        description: "test description",
        products: [productSalesChannel.id]
      })
```

The hard-coded query is a bit verbose.

**Review by srindom — CHANGES_REQUESTED** [review]

File: integration-tests/api/__tests__/admin/order.js (hunk 1 above)
Comment (adrien2p) [reply integration-tests/api/__tests__/admin/order.js:41]: Would it be alright if I create a sub describe where I would do that?

Comment (srindom) [reply integration-tests/api/__tests__/admin/order.js:41]: Yeah - have no strong opinions on how this should be structured as long as we have all existing test running without the flag :)

**Review by srindom — APPROVED** [review]
LGTM!

**Pull request merged**